realm-annotations-processor/src/main/java/io/realm/processor/ModuleMetaData.java

```diff
@@ -82,16 +75,12 @@ public boolean generate(ProcessingEnvironment processingEnv) {
                 classes = availableClasses;
             } else {
                 classes = new HashSet<ClassMetaData>();
-                for (Class<?> clazz : module.classes()) {
-                    if (!clazz.getSuperclass().toString().endsWith("RealmObject")) {
-                        Utils.error(clazz.getSimpleName() + " is not extending RealmObject. Only RealmObjects can be " +
-                                "part of a module.");
-                        return false;
-                    }
-                    ClassMetaData metadata = classMetaData.get(clazz.getName());
+                Set<String> classNames = getClassMetaDataFromModule(classElement);
+                for (String fullyQualifiedClassName : classNames) {
+                    ClassMetaData metadata = classMetaData.get(fullyQualifiedClassName);
                     if (metadata == null) {
-                        Utils.error(Utils.stripPackage(qualifiedName) + " could not be added to the module. It is not " +
-                                "possible to add classes which are part of another library.");
+                        Utils.error(Utils.stripPackage(fullyQualifiedClassName) + " could not be added to the module. " +
+                                "Only classes extending RealmObject that are part of this project can be added.");
```

Comment:
"RealmObject that are part of this project can be added" -> RealmObject, which are part of this project, can be added"